src/RadixDlt.NetworkGateway.PostgresIntegration/Queries/EntityResourcesQuery.cs

```diff
@@ -0,0 +1,447 @@
+    ORDER BY from_state_version DESC
+    LIMIT 1
+) th ON TRUE
+LEFT JOIN LATERAL (
+    (
+        SELECT *
+        FROM entity_resource_entry_definition
+        WHERE
+            entity_id = var.entity_id
+          AND resource_type = 'fungible'
+          AND from_state_version <= var.at_ledger_state
+          AND CASE WHEN var.resource_entity_id IS NOT NULL THEN resource_entity_id = var.resource_entity_id ELSE TRUE END
+          AND CASE WHEN var.descending_order THEN (from_state_version, id) <= var.resource_cursor ELSE (from_state_version, id) >= var.resource_cursor END
+        ORDER BY
+            CASE WHEN var.descending_order THEN from_state_version END DESC,
+            CASE WHEN var.descending_order THEN resource_entity_id END DESC,
+            CASE WHEN NOT var.descending_order THEN from_state_version END,
+            CASE WHEN NOT var.descending_order THEN resource_entity_id END
+        LIMIT var.fungible_resources_per_entity
+    )
+    UNION ALL
+    (
+        SELECT *
+        FROM entity_resource_entry_definition
+        WHERE
+            entity_id = var.entity_id
+          AND resource_type = 'non_fungible'
+          AND from_state_version <= var.at_ledger_state
+          AND CASE WHEN var.resource_entity_id IS NOT NULL THEN resource_entity_id = var.resource_entity_id ELSE TRUE END
+          AND CASE WHEN var.descending_order THEN (from_state_version, id) <= var.resource_cursor ELSE (from_state_version, id) >= var.resource_cursor END
+        ORDER BY
+            CASE WHEN var.descending_order THEN from_state_version END DESC,
+            CASE WHEN var.descending_order THEN resource_entity_id END DESC,
+            CASE WHEN NOT var.descending_order THEN from_state_version END,
+            CASE WHEN NOT var.descending_order THEN resource_entity_id END
+        LIMIT var.non_fungible_resources_per_entity
+    )
+) ed ON TRUE
+LEFT JOIN entities ed_entity ON ed_entity.id = ed.resource_entity_id
+LEFT JOIN LATERAL (
+    SELECT *
+    FROM entity_resource_balance_history
+    WHERE entity_id = var.entity_id AND resource_entity_id = ed.resource_entity_id AND from_state_version <= var.at_ledger_state
+    ORDER BY from_state_version DESC
+    LIMIT 1
+) ed_balance ON TRUE
+LEFT JOIN LATERAL (
+    SELECT *
+    FROM entity_resource_vault_totals_history
+    WHERE entity_id = var.entity_id AND resource_entity_id = ed.resource_entity_id AND from_state_version <= var.at_ledger_state
+    ORDER BY from_state_version DESC
+    LIMIT 1
+) ed_vault_totals ON var.use_vault_aggregation
+LEFT JOIN LATERAL (
+    SELECT *
+    FROM entity_resource_vault_entry_definition
+    WHERE
+        entity_id = var.entity_id
+      AND resource_entity_id = ed.resource_entity_id
+      AND from_state_version <= var.at_ledger_state
+      AND CASE WHEN var.descending_order THEN (from_state_version, id) <= var.vault_cursor ELSE (from_state_version, id) >= var.vault_cursor END
```

Comment:
We can obviously rename all `vaultCursor`s into `inclusiveVaultCursor`s but maybe it would be more reasonable to just document that as part of the generic guidelines proposed in the other PR?

Reply:
The generic guidelines actually used an exclusive cursor formulation. We can document it in the guidelines, yes.

My years of coding has taught me that explicitly marking inclusive / exclusive reduces off by one errors by about 80% so is worth doing; but I generally like longer / more descriptive names anyway. I'm not going to mandate this.